src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.AsyncToken.cs

```diff
@@ -20,6 +20,8 @@ public AsyncToken(AsynchronousOperationListener listener)
                 listener.Increment_NoLock();
             }
 
+            public CancellationToken BlockedOnCompletion => _listener?.BlockedOnCompletion ?? default;
```

Comment:
Is there a reason this should be conceptually associated with a token or is just for convenience because sometimes we pass around a token and not the originating listener?